lib/src/interface/common.dart

```diff
@@ -108,10 +114,24 @@ String? getExecutablePath(
       return path;
     }
   }
-  if (errorOnNull) {
-    throw ArgumentError('Failed to resolve $command to an executable.\n'
-        'workingDirectory: $workingDirectory, '
-        'candidates: ${candidates.length}');
+  if (throwOnFailure) {
+    if (foundCandidates.isNotEmpty) {
+      throw ProcessPackageExecutableNotFoundException(
+        executable,
+        message:
+            'Found candidates, but failed to resolve $executable to an executable.',
+        workingDirectory: workingDirectory,
+        candidates: candidates,
```

Comment:
This should be foundCandidates right?

Reply:
Yikes.  Yes. I'll see if I can add a test.